auxx/src/Command/Proc.hs

```diff
@@ -0,0 +1,468 @@
+        addr <- case mDistr of
+            Nothing -> makePubKeyAddressAuxx pk
+            Just distr -> return $
+                makeAddress (PubKeyASD pk) (AddrAttributes Nothing distr)
+        return $ ValueAddress addr
+    , cpHelp = "address for the specified public key. a stake distribution \
+               \ can be specified manually (by default it uses the current epoch \
+               \ to determine whether we want to use bootstrap distr)"
+    },
+
+    CommandProc
+    { cpName = "addr-hd"
+    , cpArgumentConsumer = getArg tyInt "i"
+    , cpExec = \i -> do
+        sks <- getSecretKeysPlain
+        sk <- evaluateWHNF (sks !! i)
+        addrHD <- deriveHDAddressAuxx sk
+        return $ ValueAddress addrHD
+    , cpHelp = "address of the HD wallet for the specified public key"
+    },
+
+    CommandProc
+    { cpName = "tx-out"
+    , cpArgumentConsumer = do
+        txOutAddress <- getArg tyAddress "addr"
+        txOutValue <- getArg tyCoin "value"
+        return TxOut{..}
+    , cpExec = return . ValueTxOut
+    , cpHelp = "construct a transaction output"
+    },
+
+    CommandProc
+    { cpName = "dp"
+    , cpArgumentConsumer = do
+        adpStakeholderId <- getArg (tyStakeholderId `tyEither` tyPublicKey `tyEither` tyInt) "s"
+        adpCoinPortion <- getArg tyCoinPortion "p"
+        return (adpStakeholderId, adpCoinPortion)
+    , cpExec = \(sId', cp) -> do
+        sId <- toLeft sId'
+        return $ ValueAddrDistrPart (AddrDistrPart sId cp)
+    , cpHelp = "construct an address distribution part"
+    },
+
+    CommandProc
+    { cpName = "distr"
+    , cpArgumentConsumer = getArgSome tyAddrDistrPart "dp"
+    , cpExec = \parts -> do
+        distr <- case parts of
+            AddrDistrPart sId coinPortion :| []
+              | coinPortion == maxBound ->
+                return $ SingleKeyDistr sId
+            _ -> eitherToFail $
+                 mkMultiKeyDistr . Map.fromList $
+                 map (\(AddrDistrPart s cp) -> (s, cp)) $
+                 toList parts
+        return $ ValueAddrStakeDistribution distr
+    , cpHelp = "construct an address distribution (use 'dp' for each part)"
+    },
+
+    procConst "neighbours" $ ValueSendMode SendNeighbours,
+    procConst "round-robin" $ ValueSendMode SendRoundRobin,
+    procConst "send-random" $ ValueSendMode SendRandom,
+
+    procConst "boot" $ ValueAddrStakeDistribution BootstrapEraDistr,
+
+    procConst "true" $ ValueBool True,
+    procConst "false" $ ValueBool False,
+
+    CommandProc
+    { cpName = "balance"
+    , cpArgumentConsumer = getArg (tyAddress `tyEither` tyPublicKey `tyEither` tyInt) "addr"
+    , cpExec = \addr' -> do
+        addr <- toLeft addr'
+        balance <- getBalance addr
+        return $ ValueNumber (fromIntegral . unsafeGetCoin $ balance)
+    , cpHelp = "check the amount of coins on the specified address"
+    },
+
+    CommandProc
+    { cpName = "bvd"
+    , cpArgumentConsumer = do pure ()
```

Comment:
Is `do` preset here intentionally? 🤔

Reply:
not really, just an artifact of how I wrote code (started with `do pure ()` and extended as necessary)